sdk/include/psdk/versionhelpers.h

```diff
@@ -148,9 +149,58 @@ IsActiveSessionCountLimited()
 VERSIONHELPERAPI
 IsReactOS()
 {
-    // FIXME: Find a better method!
-    WCHAR szWinDir[MAX_PATH];
-    GetWindowsDirectoryW(szWinDir, _countof(szWinDir));
-    return (wcsstr(szWinDir, L"ReactOS") != NULL);
+    BOOL bResult = FALSE;
+    BOOL bLibFree = FALSE;
+    HMODULE hMod;
+
+    hMod = GetModuleHandleW(L"shell32");
+    if (hMod == NULL)
+    {
+        hMod = LoadLibraryW(L"shell32");
+        bLibFree = (hMod != NULL);
+    }
+
+    if (hMod)
+    {
+        DWORD dwLen;
+        WCHAR szFullPath[MAX_PATH];
+
+        dwLen = GetModuleFileNameW(hMod, szFullPath, MAX_PATH);
+        if (dwLen)
+        {
+            DWORD dwSize;
+
+            szFullPath[dwLen] = 0; // required by Windows XP
+            dwSize = GetFileVersionInfoSizeW(szFullPath, NULL);
+            if(dwSize)
+            {
+                BOOL bProduct;
+                UINT cchVer = 0;
+                LPCWSTR lszValue = NULL;
+                HANDLE hMem = NULL;
+                LPVOID VerInfo = NULL;
+
+                hMem = GlobalAlloc(GMEM_MOVEABLE, dwSize);
```

Comment:
Prefer using HeapAlloc/HeapFree.

Reply:
Do the GlobalLock and GlobalUnlock functions work or do alternative functions like HeapLock and HeapUnlock exist?

Reply:
Why do you need "lock"/"unlock" stuff?